### PR TITLE
DHFPROD-4621: Only upload data once

### PR DESCRIPTION
**Pull request by ryanjdew**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: one-ui/ui/src/components/load-data/new-data-load-dialog/new-data-load-dialog.test.tsx [hunk 1]
```diff
@@ -1,9 +1,89 @@
...
 
-describe('New/Edit data load component', () => {
-  it('should render correctly', () => {
-    shallow(<NewDataLoadDialog data/>);
+jest.mock('axios');
+
+axiosMock.mockImplementation(jest.fn(() => Promise.resolve({ status: 200, data: {} })));
+axiosMock.post.mockImplementation(jest.fn(() => Promise.resolve({ status: 200, data: {} })));
+axiosMock.get.mockImplementation(jest.fn(() => Promise.resolve({ status: 200, data: {} })));
+
```
Comment (bsrikan) [inline one-ui/ui/src/components/load-data/new-data-load-dialog/new-data-load-dialog.test.tsx]: This looks good @ryanjdew . Can we clear the mocks after a test is done?

Comment (ryanjdew) [reply one-ui/ui/src/components/load-data/new-data-load-dialog/new-data-load-dialog.test.tsx]: @bsrikan Do you mean something like below to clear the counts?
```
afterEach(() => {
  jest.clearAllMocks();
});
```

Comment (bsrikan) [reply one-ui/ui/src/components/load-data/new-data-load-dialog/new-data-load-dialog.test.tsx]: Right. So it doesnt bleed into other tests that may use those mocks.

**Comment by SameeraPriyathamTadikonda** [comment]
@bsrikan/ @wooldridge / @xnikhil08 Please check the reviewed tests checkbox if tests are reviewed.

File: one-ui/ui/src/components/load-data/new-data-load-dialog/new-data-load-dialog.test.tsx (hunk 1 above)
Comment (ryanjdew) [reply one-ui/ui/src/components/load-data/new-data-load-dialog/new-data-load-dialog.test.tsx]: @bsrikan Ok. Added the snippet above and rebased.

**Pull request merged**